Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```diff
@@ -102,103 +94,79 @@ public CONNECTSamlAssertionValidator(PropertyAccessor propertyAccessor) {
      * @throws WSSecurityException the wS security exception
      */
     @Override
-    protected void validateAssertion(AssertionWrapper assertion) throws WSSecurityException {
+    protected void validateAssertion(final SamlAssertionWrapper assertion) throws WSSecurityException {
         if (assertion.getSaml1() != null) {
-            ValidatorSuite schemaValidators = org.opensaml.Configuration.getValidatorSuite("saml1-schema-validator");
-            ValidatorSuite specValidators = org.opensaml.Configuration.getValidatorSuite("saml1-spec-validator");
-            try {
-                schemaValidators.validate(assertion.getSaml1());
-                specValidators.validate(assertion.getSaml1());
-            } catch (ValidationException e) {
-                LOG.debug("Saml Validation error: " + e.getMessage(), e);
-                throw new WSSecurityException(WSSecurityException.FAILURE, "invalidSAMLsecurity");
-            }
+            assertion.validateSignatureAgainstProfile();
         } else if (assertion.getSaml2() != null) {
-            List<ValidatorSuite> validators = new LinkedList<>();
-            validators.add(org.opensaml.Configuration.getValidatorSuite("saml2-core-schema-validator"));
-            validators.addAll(getSaml2SpecValidators());
-
-            for (AuthzDecisionStatement auth : assertion.getSaml2().getAuthzDecisionStatements()) {
-                if (StringUtils.isBlank(auth.getResource())) {
-                    auth.setResource(TEMP_RESOURCE_FOR_VALIDATION);
+            final Issuer issuer = assertion.getSaml2().getIssuer();
+            validateIssuer(issuer);
+            try {
+                // need to process custom saml validator
+                for (final Saml2ExchangeAuthFrameworkValidator validator : getSaml2SpecValidators()) {
+                    validator.validateAssertion(assertion);
                 }
+            } catch (final WSSecurityException e) {
+                LOG.error("Saml Validation error: " + e.getMessage(), e);
```

Comment:
Follow-up:  Here's a neat trick you can use to look for uses of "getMessage" (or other Strings/regexes) in code that has been changed as part of a commit:

git diff-index CONNECT -G getMessage --name-only

This will list only the files where you've added or changed a "getMessage" line, but NOT files where it already existed and you left it unchanged.